src/Compilers/Core/Portable/ReferenceManager/CommonReferenceManager.Resolution.cs

```diff
@@ -52,23 +52,54 @@ protected struct ResolvedReference
         {
             private readonly MetadataImageKind _kind;
             private readonly int _index;
-            private readonly ImmutableArray<string> _aliases;
+            private readonly ImmutableArray<string> _aliasesOpt;
+            private readonly ImmutableArray<string> _recursiveAliasesOpt;
 
-            public ResolvedReference(int index, MetadataImageKind kind, ImmutableArray<string> aliases)
+            // uninitialized aliases
+            public ResolvedReference(int index, MetadataImageKind kind)
+                : this(index, kind, default(ImmutableArray<string>), default(ImmutableArray<string>))
+            {
+            }
+
+            // initialized aliases
+            public ResolvedReference(int index, MetadataImageKind kind, ImmutableArray<string> aliasesOpt, ImmutableArray<string> recursiveAliasesOpt)
             {
                 Debug.Assert(index >= 0);
+                Debug.Assert(_aliasesOpt.IsDefault || _recursiveAliasesOpt.IsDefault);
 
                 _index = index + 1;
                 _kind = kind;
-                _aliases = aliases;
+                _aliasesOpt = aliasesOpt;
+                _recursiveAliasesOpt = recursiveAliasesOpt;
             }
 
-            public ImmutableArray<string> Aliases
+            private bool IsUninitialized => _aliasesOpt.IsDefault && _recursiveAliasesOpt.IsDefault;
+
+            /// <summary>
+            /// Aliases that should be applied to the referenced assembly. 
+            /// Empty array means {global} (all namespaces and global types are accessible without qualification).
```

Comment:
It should rather say "all global namespaces and types are accessible without qualification"